### DISCUSSION TITLE
Districts do not import correctly

### DISCUSSION THREAD

**Discussion by YodelingCowboys** [code4recovery/12-step-meeting-list#509] [Q&A] [answered]
I did an export of lincaa.org.  I added "District XX" to every record in the .csv file.  I made sure that every "District XX" was added to TSML.  I imported the .csv file using the option "delete all meetings, locations, groups, districts, and regions".  Almost no meetings have any district information.  

I did an export of Area 41 meetings.  Edited the ,csv file to delete all Lincoln meetings.  Copied all Lincoln meetings from the lincaa.org export and pasted them to the Area 41 .csv.  I checked the Area 41 .csv and every row had "District XX" in the "District" column.  I imported the Area 41 .csv.  Very few Lincoln meetings have any District information.

I think there is a bug somewhere.  What can I do to help you find and correct it?

Thanks.

**Comment by tech2serve** [comment]
> I think there is a bug somewhere.  What can I do to help you find and correct it?
> 
> 
> 
> Thanks.

One of us will try and duplicate what you're experiencing on a dev site. It may be a few days before I have time, but I'll try.

Given that we have a number of users who successfully do routine imports of their data, I believe we should focus on your site and any errors you can see.

Please open a dev console/JavaScript console for your browser. How to do this depends on your browser, and you should use your favorite search engine to figure that out. Afterwards, try the import again and see if any errors get displayed.

Also, you can send a copy of your csv file to devs@code4recovery.org so we can inspect the sheet.

**Reply by YodelingCowboys** [reply]
When I first updated these meetings in TSLM, I changed all to "individual meeting" rather than "part of a group" because, to me, a distinct meeting at a distinct location is not "part of a group."  I added the district information and updated again.  The district info disappeared.

I tried to change a meeting back to "part of a group" and added the district info again.  After clicking "Update", the meeting was again back to "individual meeting" with no district info.

I edited the .csv file to add the district info, trashed TSML, reinstalled TSML, added the Districts to TSML, and imported the .csv using the option "delete all meetings, locations, groups, districts, and regions".  All meetings were again set to "individual meeting" with no district info.

I created "District Online" for online-only meetings since without an address, I do not know in which district they belong.

It appears that the setting "individual meeting" does not update when changed to "part of a group."  And I do not understand why an "individual meeting" does not save district info when updated.

I am creating a staging site where I can see if it works the same as my live site.

Thanks for your help.  I really appreciate the work you guys do supporting this plugin.

[Lincaa_asof_20210823.csv](https://github.com/code4recovery/12-step-meeting-list/files/7069321/Lincaa_asof_20210823.csv)

**Reply by YodelingCowboys** [reply]
I have created a simply staging site, installed TSML, added the districts, imported the Lincaa_asof_20210823.csv file, and the meetings are "individual meetings" with no district information.

I will contact my host provider (DreamHost.com) to see if the problem might be in their caching.  My staging site is lincaa.dreamhosters.com.

**Comment by YodelingCowboys** [comment, marked as answer]
SOLVED!!!!

I figured out the undocumented feature.  For a meeting to be "part of a
group", there must be a "Group Name".  Understandable for groups that have
multiple meetings.

But for a group that only meets once a week, it must also be "part of a
group" with a "Group Name" in order to have any district.

Perhaps the "Meeting Name" should auto-populate the "Group Name."

Question is why can't an "individual meeting" be part of a district?  This
I think is a bug.

**Reply by tech2serve** [reply]
So glad you got it figured out. 

As is often the case, the data being entered is important. We do recognize there are areas in the UI that would benefit from more feedback to the user when things aren't quite right. CSV files are tricky because this is harder to accomplish. This is why I often suggest users create at least a couple of meetings directly in the system so they can export that to a CSV and see the template the system really expects. I had thought you were doing this. 

I'll still try to look closer at this later and see if I can duplicate some of your experience. Typically, when a piece of information is missing, the meeting won't publish. 

As far as groups vs individual meetings, I'll tell you that I have considerable experience traveling across the country, and how local service organizations think about this differs greatly. In my experience (and how I was trained by an early sponsor) a Group is the basic part of AA's service structure, with Districts being at the next level. In this regard, a meeting is not by itself a part of the service organization (two or more drunks together can call themselves a meeting). I will say that often people use the terms "meeting" and "group" interchangeably. 

TSML also provides the user the option to establish Regions, and this is also a useful tool for those areas where the AA culture doesn't focus on the service structure, or it is a large metropolitan area that has both.

As you can probably see, we try to have TSML be so that it can flex to these different models. 

Glad you got it figured out. Please mark your answer (discovering the need to add a Group name) as the Answer and we'll close this out.
